xarray/core/coordinates.py

```diff
@@ -399,5 +399,7 @@ def remap_label_indexers(
             # drop coordinates found in indexers since .sel() already
             # ensures alignments
             coords = {k: var for k, var in v._coords.items() if k not in indexers}
-            pos_indexers[k] = DataArray(pos_indexers[k], coords=coords, dims=v.dims)
+            pos = DataArray(pos_indexers[k], coords=coords, dims=v.dims)
+            if pos.dtype.kind == "i":
+                pos_indexers[k] = pos
```

Comment:
Why should this only be done for integer dtypes? What happens with non-integer positional indexers?

I'm a little puzzled by how this is related to the other changes in this PR. It might make sense to add some specific tests related to whatever is being fixed here?

Reply:
Thanks for reviewing.
This meant to avoid wrapping a slice by a DataArray.
Updated to a cleaner logic.